blocks/soundcloud/soundcloud.js

```diff
@@ -0,0 +1,46 @@
+const SOUNDCLOUD = {
+  origin: 'https://w.soundcloud.com',
+  url: 'https://api.soundcloud.com/tracks/',
+};
+
+const calculateIframeHeight = (text) => {
+  // Extract the height value from the iframe HTML
+  const heightRegex = /height="(\d+)"/;
+  const heightMatch = text.match(heightRegex);
+  if (heightMatch && heightMatch[1]) {
+    return parseInt(heightMatch[1], 10);
+  }
+  return 0;
+};
+
+export default async function decorate(block) {
+  const iframes = Array.from(block.querySelectorAll('.soundcloud > div'))
```

Comment:
you could save some CPU cycles by using `reduce`. would look something like this (uses separate function to create iframe, see next comment):
```
export default function decorate(block) {
  const divElements = Array.from(block.querySelectorAll('.soundcloud > div'));
  divElements.reduce((iframes, element) => {
    const innerDiv = element.children[1];
    const text = innerDiv.textContent;
    const match = text.match(/src="(.*?)"/);
    if (match && /<iframe.*?<\/iframe>/.test(text)) {
      const url = new URL(decodeURIComponent(match[1]));
      const urlParam = new URLSearchParams(url.search).get('url');
      if (url.origin === SOUNDCLOUD.origin && urlParam?.startsWith(SOUNDCLOUD.url)) {
        iframes.push({ element, innerDiv, url, text });
      }
    }
    return iframes;
  }, []).forEach(({ element, innerDiv, url, text }) => {
    const height = calculateIframeHeight(text);
    const soundCloudFrame = createSoundCloudIframe(url, height);
    innerDiv.innerHTML = '';
    innerDiv.appendChild(soundCloudFrame);
    element.classList.add('soundcloud-col');
  });
}
```